ui/docs/Architecture.md

```diff
@@ -0,0 +1,95 @@
+# Architecture
+
+The UI for the starter application is implemented using React, as well as a set
+of other dependancies, such as Webpack for build and Jest as a test runner. The
+UI however is one part of the whole starter application.
+
+The following documentation will describe the topology of the starter 
+application when developing the UI, as well as when it is deployed as a
+part of the fully built starter application. In addition, it will describe
+the composition of React components used to build up the UI.
+
+## Topology summary
+
+### When deployed
+
+The starter application's topology when run by a user will resemble the 
+following:
+
+![Starter app topology](./ArchImages/Topology.jpg)
+
+When running the Jar file, a Vertx server will be started, which in turn will
+launch three verticles. One verticle will handle records being produced to
+Kafka, one verticle will handle records being consumed from Kafka, and finally
+the third verticle deployed will server the UI when requested, along with all
+supporting content, such as fonts and images. The producer and consumer 
+verticles are configured to access Kafka via configuration provided by the user
+at start up.
+
+Once running, the user access the the endpoint serving UI content. This returns
+the index.html, which will in turn contain script and source tags which will
+pull all required styling and JavaScript to run the UI. The index.html will 
+also contain a `meta` tag, which will contain configuration the UI will need to
+operate with. It would for example contain the name of the topic this 
+application has been configured to produce and consume from.
+
+Once all required JS and style resources have been retrieved, the UI's `App`
+component (detailed below) will bootstrap the rest of the UI. This will in turn
+cause two websocket connections to be created - one for the producer, and one
+for the consumer. These websockets are then driven and interacted with as the
+user uses the application.
+
+### When developing the UI
+
+When the UI is developed, the topology used is different versus a normal 
+deployment:
+
+![UI development/test topology](./ArchImages/TopologyDev.jpg)
+
+In this case, the UI, and all its assets, will be served via webpack-dev-server.
+webpack-dev-server enables faster UI development by offering capabilities such
```

Comment:
Should webpack-dev-server be in backticks?

Reply:
Updated